packages/gatsby-theme-docs/configure-theme.js

```diff
@@ -0,0 +1,25 @@
+const pkg = require('./package.json');
+
+function configureThemeWithAddOns(themeOptions = {}) {
+  const { addOnPlugins, ...coreThemeOptions } = themeOptions;
+
+  if (!addOnPlugins || !Array.isArray(addOnPlugins)) {
```

Comment:
Why would the function not work if no add on plugins are configured?  We could default the option to an empty array here, that would make copy / pasting and modifying site configurations (e.g. commenting out that line to test) more robust.

Reply:
Sure, if you prefer so. The check was meant to prevent the user to use the function without passing any add-on.